### PR TITLE
feat: read file during reconstruction

### PR DESCRIPTION
**Pull request by rphmeier**

### CODE REVIEW

**Comment by rphmeier** [comment]
* **#34** <a href="https://app.graphite.dev/github/pr/thrumdev/bitbox/34?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#33** <a href="https://app.graphite.dev/github/pr/thrumdev/bitbox/33?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#32** <a href="https://app.graphite.dev/github/pr/thrumdev/bitbox/32?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#35](https://github.com/thrumdev/bitbox/pull/35) <a href="https://app.graphite.dev/github/pr/thrumdev/bitbox/35?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#31** <a href="https://app.graphite.dev/github/pr/thrumdev/bitbox/31?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rphmeier and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by pepyakin — DISMISSED** [review]
Perhaps not immediatelly obvious and not stated anywhere explicitly (maybe briefly in meta.rs) but the first page of BBN and LN storages are reserved. The reason is two-fold:

1. to have a notion of NULL pointer for specifying the end of a linked-list e.g. for freelist. (It is used for LN storage and we are discussing it for BBN storage)
2. just to reserve some space for some data. I am not entirely sure under which circumstances we would need it, given that we already have an entire metadata file dedicated for that stuff, but who knows.

so for this code, this basically implies we should just skip the first page.

File: src/beatree/ops/reconstruction.rs [hunk 1]
```diff
@@ -20,7 +20,8 @@ use crate::beatree::{
 };
 
 // 256K
-type SeqReadChunk = [u8; BRANCH_NODE_SIZE * 64];
+const READ_CHUNK_SIZE: usize = BRANCH_NODE_SIZE * 64;
```
Comment (pepyakin) [inline src/beatree/ops/reconstruction.rs:23]: Nit: or `SEQ_READ_CHUNK_LEN` ? : )

Comment (pepyakin) [inline src/beatree/ops/reconstruction.rs:23]: Nit: or `SEQ_READ_CHUNK_LEN` ? : )

File: src/beatree/ops/reconstruction.rs [hunk 2]
```diff
@@ -20,7 +20,8 @@ use crate::beatree::{
...
+type SeqReadChunk = [u8; READ_CHUNK_SIZE];
```
Comment (pepyakin) [inline src/beatree/ops/reconstruction.rs]: Nit: or `SEQ_READ_CHUNK_LEN`

**Comment by rphmeier** [comment]
Thanks for clarifying. I've updated the PR to skip the first page.

**Comment by rphmeier** [comment]
### Merge activity

* **Jul 22, 12:57 PM EDT**: @rphmeier started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/thrumdev/bitbox/34).
* **Jul 22, 1:05 PM EDT**: [Graphite](https://app.graphite.dev/github/pr/thrumdev/bitbox/34) rebased this pull request as part of a merge.
* **Jul 22, 1:06 PM EDT**: @rphmeier merged this pull request with [Graphite](https://app.graphite.dev/github/pr/thrumdev/bitbox/34).

**Pull request merged**